### DISCUSSION TITLE
Autopilot in Unreal Engine

### DISCUSSION THREAD

**Discussion by MikeKilo** [JSBSim-Team/jsbsim#973] [Support requests]
I have a project running with the JSBSimMovement component, with a model of Cessna 172 X and all is working as it should.
I want to use the c172ap inside the unreal engine, so I can direct it to waypoints and control it as a UAV.
Is there any way to do it with the current plugin, or should a different plugin be made in order to access the autopilot capabilities?

**Comment by gallonmate** [comment]
If you use blueprints you can use these blueprint nodes to send any command to JSBSim. 
https://github.com/JSBSim-Team/jsbsim/tree/9f92a0603498ff09dd84e77b6e6022b866fcb50f/UnrealEngine#extended-commands-and-properties

You can look through these scripts to find the various autopilot commands which you can use in the blueprint nodes. https://github.com/JSBSim-Team/jsbsim/tree/master/scripts

So for example this script which uses the c172x:
https://github.com/JSBSim-Team/jsbsim/blob/ae4652a268d0475dbf93d47a32ab7ddf4120aaef/scripts/c172_cruise_8K.xml#L80-L87

You can see the commands which you can enter in the blueprint node as:
**Property:** ap/heading_setpoint **Value**: 200
**Property:** ap/heading_hold **Value**: 1
**Property:** ap/altitude_setpoint **Value**: 4000
**Property:** ap/altitude_hold **Value**: 1

A value of 1 and 0 will enable/disable the appropriate systems.
There are other commands if you search and some examples from other aircraft might be useful as well.

And a caution that using incorrect property names or properties that haven't been loaded yet by jsbsim will crash UE5. So if UE5 crashes when using commands, then this is why.

**Comment by MikeKilo** [comment]
Thank you very much, everything works like you said!
